hoomd/tune/attr_tuner.py

```diff
@@ -0,0 +1,332 @@
+from abc import ABCMeta, abstractmethod
+
+
+class _TuneDefinition(metaclass=ABCMeta):
+    """Internal class for defining y = f(x) relations
+
+    This class is designed to allow for tuning x to achieve a specified value
+    for y over a domain T. It abstracts over getting and setting x and getting
+    y. The class also provides helper functions for ensuring x is always set to
+    a value within the specified domain.
+    """
+    def __init__(self, target, domain=None):
+        self.domain = domain
+        self._target = target
+
+    def in_domain(self, value):
+        """Check whether a value is in the domain.
+
+        Args:
+            value (``any``): A value that can be compared to the minimum and
+                maximum of the domain.
+
+        Returns:
+            bool: Whether the value is in the domain of x.
+        """
+        if self.domain is None:
+            return True
+        else:
+            lower_bound, upper_bound = self.domain
+            return ((lower_bound is None or lower_bound <= value)
+                    and (upper_bound is None or value <= upper_bound))
+
+    def wrap_into_domain(self, value):
```

Comment:
`wrap` implies that values will wrap around periodic boundary conditions. Use `clamp` instead.